solidago/src/solidago/mehestan/individual/continuous_bradly_terry.py

```diff
@@ -0,0 +1,156 @@
+    def __init__(self, r_max, alpha=DEFAULT_ALPHA):
+        """
+        Initialize Continous Bradley Terry (CBT) to compute individual scores
+
+        Parameters
+        ----------
+        r_max: Maximum absolute score of a comparison
+        alpha: Regularization term
+        """
+        self.alpha = alpha
+        self.r_max = r_max
+
+    def coordinate_descent(self, coord_to_subset, initial_scores):
+        n_alternatives = len(coord_to_subset)
+        unchanged = set()
+        to_pick = []
+
+        def pick_next_coordinate():
+            nonlocal to_pick
+            if len(to_pick) == 0:
+                to_pick = list(range(n_alternatives))
+                random.shuffle(to_pick)
+            return to_pick.pop()
+
+        theta = initial_scores
+        while len(unchanged) < n_alternatives:
+            coord = pick_next_coordinate()
+            if coord in unchanged:
+                continue
+            indices, r_ab = coord_to_subset[coord]
+            old_theta_a = theta[coord]
+            theta_b = theta[indices]
+            new_theta_a = coordinate_optimize(
+                r_ab, theta_b, precision=EPSILON / 10, alpha=self.alpha
+            )
+            theta[coord] = new_theta_a
+            if abs(new_theta_a - old_theta_a) < EPSILON:
+                unchanged.add(coord)
+            else:
+                unchanged.clear()
+        return theta
+
+    def compute_individual_scores(self, scores: pd.DataFrame, initial_entity_scores=None):
+        scores = scores[["entity_a", "entity_b", "score"]]
+        scores_sym = pd.concat(
+            [
+                scores,
+                pd.DataFrame(
+                    {
+                        "entity_a": scores.entity_b,
+                        "entity_b": scores.entity_a,
+                        "score": -1 * scores.score,
+                    }
+                ),
+            ]
+        )
+        r = scores_sym.pivot(index="entity_a", columns="entity_b", values="score") / self.r_max
+        r_values = r.to_numpy()
+        n_entities = len(r_values)
+        coord_to_subset = {}
+        for coord in range(n_entities):
+            r_ab = r_values[coord, :]
+            indices = (~np.isnan(r_ab)).nonzero()
+            coord_to_subset[coord] = (indices, r_ab[indices])
+
+        if initial_entity_scores is None:
+            initial_scores = np.zeros(n_entities)
+        else:
+            initial_scores = pd.Series(initial_entity_scores, index=r.index)
+            initial_scores.fillna(0.0, inplace=True)
+            initial_scores = initial_scores.to_numpy()
+        theta_star_numpy = self.coordinate_descent(coord_to_subset, initial_scores=initial_scores)
+        delta_star_numpy = np.zeros(len(theta_star_numpy))
+        for idx in range(len(theta_star_numpy)):
+            indices, _r_ab = coord_to_subset[idx]
+            delta_star_numpy[idx] = Delta_theta(theta_star_numpy[indices])
```

Comment:
Is it expected that the uncertainty for video_A depends only on the individual scores of the compared videos to video_A.

The formula seems to use `theta_ab`, maybe you should do the difference? 
```suggestion
            delta_star_numpy[idx] = Delta_theta(theta_star_numpy[indices] - theta_star_numpy[idx])
```
?

Reply:
Yes indeed. Fixed in 99ba5e5